test/integration/action_execution.go

```diff
@@ -379,7 +379,7 @@ func createDeployResource(client *kubeclientset.Clientset, dep *appsv1.Deploymen
 
 // This can also be bootstrapped from a test resource directory
 // which holds yaml files.
-func depSingleContainerWithResources(namespace, claimName string, replicas int32, withVolume, withGCLabel, paused bool) *appsv1.Deployment {
+func depSingleContainerWithResources(namespace, claimName string, replicas int32, withVolume, withGCLabel, paused bool, nodeName string, pullSecrets []corev1.LocalObjectReference) *appsv1.Deployment {
```

Comment:
I don't think you need to add this `pullSecrets` parameter, please refer to `depMultiContainerWithResources`, you can directly add the `imagePullSecret` in pod's Spec even that secret might not exist at all. if we can do this, I guess we don't need the function `DockerImagePullSecretNames` any more.